rstudio/rsconnect-jupyter
# API key rework

### Description

This PR moves entry of the API key from the Publish dialog to the Add Server dialog. API key storage now uses the Jupyter plugin config storage (previously saved API keys are not migrated).

Connected to #177 

### Testing Notes / Validation Steps
* Open a notebook you've published previously (on an older version of the plugin). Click the button to publish. You should see the Add Server dialog, since you don't have any saved API keys.
* Add a server, but cancel the publish dialog. Open it again. The server should be present and you should be able to publish.
* Try to add a server with an invalid URL; the usual error about being unable to verify the server should be displayed.
* Try to add a server with an invalid API key; the error should reflect that the key was invalid.
